### PR TITLE
Move arbitrary self types's tests into ui/self

### PR DESCRIPTION
**Pull request by taiki-e**
https://github.com/rust-lang/rust/pull/60944#discussion_r285362006

r? @Centril

### CODE REVIEW

File: src/test/ui/self/arbitrary_self_types_pointers_and_wrappers.rs [hunk 1]
```diff
@@ -1,3 +1,4 @@
+// run-pass
```
Comment (petrochenkov) [inline src/test/ui/self/arbitrary_self_types_pointers_and_wrappers.rs]: Please don't move run-pass tests into ui directory, or move and make them compile-pass if they don't really require running.

Comment (Centril) [reply src/test/ui/self/arbitrary_self_types_pointers_and_wrappers.rs]: @petrochenkov We're trying to retire the `run-pass` directory altogether; It's pretty bad organization to have some tests for some feature in `test/run-pass/foo` and some in `test/ui/foo`. But yeah, make it `compile-pass` if it doesn't need to be actually run.

**Comment by Centril** [comment]
@taiki-e Can you please move the other files in the `run-pass/self` directory also?

File: src/test/ui/self/arbitrary_self_types_pointers_and_wrappers.rs (hunk 1 above)
Comment (petrochenkov) [reply src/test/ui/self/arbitrary_self_types_pointers_and_wrappers.rs]: @Centril 
>We're trying to retire the run-pass directory altogether

This was tried, but reverted, see https://github.com/rust-lang/rust/pull/54530.

Comment (Centril) [reply src/test/ui/self/arbitrary_self_types_pointers_and_wrappers.rs]: Seems like an unfortunate move to me. Something should be done about the current chaos in the `src/test` directory. One shouldn't have to look in different places for tests pertaining to one specific interest. That is especially annoying when writing stabilization reports and whatnot.

For now, let's move ahead with moving these specific tests.

**Comment by taiki-e** [comment]
@Centril How should I handle files of the same name?

File: src/test/ui/self/explicit-self-closures.rs [hunk 2]
```diff
@@ -1,4 +1,4 @@
-// run-pass
+// compile-pass
```
Comment (taiki-e) [inline src/test/ui/self/explicit-self-closures.rs]: Changed `run-pass` test with empty main function to `compile-pass`.

**Pull request merged**